### PR TITLE
Auto Google Maps

### PR DESCRIPTION
**Pull request by fhamon**
Simple module to add maps in a page

Features:
- Custom styling via a script tag (application/json)
- `N` markers w/ or w/o infobox in it
- Custom icons per markers

### CODE REVIEW

File: src/modules/auto-google-map.js [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+
+	var createMarker = function (element, mapComponent) {
+		var lat = element.attr('data-lat');
+		var lng = element.attr('data-lng');
+		var iconSrc = element.attr('data-icon');
+		var iconWidth = parseInt(element.attr('data-icon-width'), 10) || DEFAULT_MARKER_WIDTH;
+		var iconHeight = parseInt(element.attr('data-icon-height'), 10) || DEFAULT_MARKER_HEIGHT;
+
+		if (!!App.device.internetexplorer) {
+			iconSrc = null;
```
Comment (nitriques) [inline src/modules/auto-google-map.js]: why ?

File: src/modules/auto-google-map.js [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+				p2: {
+					x: iconWidth / 2,
+					y: iconHeight
+				}
+			},
+			zIndex: 50
+		};
+
+		if (!iconSrc) {
+			delete config.iconImage;
```
Comment (nitriques) [inline src/modules/auto-google-map.js]: Set the property to undefined, then assign something do it instead

File: src/modules/auto-google-map.js [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+		initAllMap();
+		ready = true;
+	};
+
+	var actions = function () {
+		return {
+			site: {
+				loaded: onSiteLoaded
+			},
+			page: {
```
Comment (nitriques) [inline src/modules/auto-google-map.js]: We should destroy before leaving, if the page gets removed.

Comment (fhamon) [reply src/modules/auto-google-map.js]: The api does not include a destory method, removing the dom will do the job.

File: src/modules/auto-google-map.js (hunk 1 above)
Comment (fhamon) [reply src/modules/auto-google-map.js]: I'm not sure, but I think it has to do with svg images as pins.

File: src/modules/auto-google-map.js [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+
+		if (!ready) {
+			return;
+		}
+
+		scope.find(sels.ctn).each(function () {
+			var t = $(this);
+
+			if (!!t.find(sels.map).data('googleMapComp')) {
+				return;
```
Comment (PascalPiche) [inline src/modules/auto-google-map.js]: maybe add a log of faillure?

File: src/modules/auto-google-map.js [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+						markers.each(function () {
+							createMarker($(this), comp);
+						});
+					}
+				});
+
+				// Store comp
+				t.data('googleMapComp', comp);
+				comp.init(t);
+			}
```
Comment (PascalPiche) [inline src/modules/auto-google-map.js]: and a else statement with a log for no lat,lng found

File: src/modules/auto-google-map.js [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+		ready = true;
+	};
+
+	var onArticleEntering = function () {
+		initAllMap();
+	};
+
+	var onSiteLoaded = function () {
+		initAllMap();
+		ready = true;
```
Comment (PascalPiche) [inline src/modules/auto-google-map.js]: we should put ready before trying to initAllMap

Comment (fhamon) [reply src/modules/auto-google-map.js]: No it's to counter the race between page.enter and site.loaded

I need both events to have trigged and only then initing all maps. (page.enter can be trigged before or after site.loaded)

File: src/modules/auto-google-map.js (hunk 4 above)
Comment (fhamon) [reply src/modules/auto-google-map.js]: 968a8dd3bad58ace3687aa5cbc0508883a3e41b8

File: src/modules/auto-google-map.js (hunk 5 above)
Comment (fhamon) [reply src/modules/auto-google-map.js]: 968a8dd3bad58ace3687aa5cbc0508883a3e41b8

File: src/modules/auto-google-map.js (hunk 1 above)
Comment (PascalPiche) [reply src/modules/auto-google-map.js]: Ping @nitriques

Comment (nitriques) [reply src/modules/auto-google-map.js]: Well @fhamon  @PascalPiche we need to be sure.... no ?

File: src/modules/auto-google-map.js (hunk 2 above)
Comment (nitriques) [reply src/modules/auto-google-map.js]: @fhamon ping!

Comment (fhamon) [reply src/modules/auto-google-map.js]: out daaaaaaated

File: src/modules/auto-google-map.js (hunk 3 above)
Comment (nitriques) [reply src/modules/auto-google-map.js]: > The api does not include a destory method,

You mean Gmaps API ?

I that case, we could at least remove the event handlers before removing from the dom ?

see https://stackoverflow.com/questions/10485582/what-is-the-proper-way-to-destroy-a-map-instance

Comment (nitriques) [reply src/modules/auto-google-map.js]: @fhamon ping!

**Comment by nitriques** [comment]
@fhamon I still have 2 open comments on this ;)

File: src/modules/auto-google-map.js (hunk 3 above)
Comment (fhamon) [reply src/modules/auto-google-map.js]: > The official answer is you don't.

Comment (nitriques) [reply src/modules/auto-google-map.js]: Yeah I can't agree with that...

**Comment by fhamon** [comment]
@nitriques done please merge if ok

**Pull request merged**